plugins/inputs/unbound/unbound.go

```diff
@@ -140,32 +149,59 @@ func (s *Unbound) Gather(acc telegraf.Accumulator) error {
 		value := cols[1]
 
 		// Filter value
-		if filter_excluded.Match(stat) {
+		if filterExcluded.Match(stat) {
 			continue
 		}
 
-		field := strings.Replace(stat, ".", "_", -1)
-
-		fields[field], err = strconv.ParseFloat(value, 64)
+		fieldValue, err := strconv.ParseFloat(value, 64)
 		if err != nil {
-			acc.AddError(fmt.Errorf("Expected a numerical value for %s = %v\n",
+			acc.AddError(fmt.Errorf("Expected a numerical value for %s = %v",
 				stat, value))
+			continue
 		}
+
+		// is this a thread related value?
+		if s.ThreadAsTag && strings.HasPrefix(stat, "thread") {
+			// split the that
+			statTokens := strings.Split(stat, ".")
```

Comment:
Check that it split correctly before indexing.

Reply:
Done! - I think :)